app/models/host/managed.rb

```diff
@@ -559,8 +559,11 @@ def inherited_attributes
   end
 
   def set_compute_attributes
-    return unless compute_profile_present?
-    self.compute_attributes = compute_resource.compute_profile_attributes_for(compute_profile_id)
+    if compute_profile_present?
+      self.compute_attributes = compute_resource.compute_profile_attributes_for(compute_profile_id)
+    elsif compute_resource
```

Comment:
Added check on compute resource presence when setting the compute attributes (with pure `else`, the bare-metal test was failing)